lib/spack/env/cc

```diff
@@ -162,7 +162,7 @@ fi
 # It doesn't work with -rpath.
 # This variable controls whether they are added.
 add_rpaths=true
-if [[ mode == ld && $OSTYPE == darwin* ]]; then
+if [[ $mode == ld && "$SPACK_SHORT_SPEC" =~ "darwin" ]]; then
```

Comment:
THIS IS PROBABLY WRONG. (Sorry for shouting, but I see that this has already been merged.) Quotes prevent the variable expansion here.

`[[ ]]` has very different quoting rules from `[ ]`.

Given that the if condition probably never succeeded (as it was wrong before as well, missing `$` sign), I wonder whether it is actually necessary?